cranelift/module/src/module.rs

```diff
@@ -627,22 +708,14 @@ impl<M: Module> Module for &mut M {
         (**self).declare_anonymous_data(writable, tls)
     }
 
-    fn declare_func_in_func(&self, func: FuncId, in_func: &mut ir::Function) -> ir::FuncRef {
+    fn declare_func_in_func(&mut self, func: FuncId, in_func: &mut ir::Function) -> ir::FuncRef {
         (**self).declare_func_in_func(func, in_func)
     }
 
     fn declare_data_in_func(&self, data: DataId, func: &mut ir::Function) -> ir::GlobalValue {
         (**self).declare_data_in_func(data, func)
     }
 
-    fn declare_func_in_data(&self, func: FuncId, ctx: &mut DataContext) -> ir::FuncRef {
-        (**self).declare_func_in_data(func, ctx)
-    }
-
-    fn declare_data_in_data(&self, data: DataId, ctx: &mut DataContext) -> ir::GlobalValue {
-        (**self).declare_data_in_data(data, ctx)
-    }
```

Comment:
These haven't yet been re-added.

Reply:
@bnjbvr?

Reply:
Ah yeah, seems I added it back in one implementation, but not in the trait. Is that something that you were using, @bjorn3? I could add it back later, if required

Reply:
This change means that `declare_func_in_data` and `declare_data_in_data` are no longer forwarded to `FooModule` when you call them on a `&mut FooModule`, which is incorrect if those functions have been overwritten by `FooModule`.

Reply:
I see; it's likely subpar design if the default implementation is incorrect for some implementations by default, but that's a different discussion to have :) Will post a small patch for this chunk.